controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,484 @@
+		// ownerReference on it as its cluster scoped resource
+		cephfsCSIDriver := templates.CephFSCSIDriver.DeepCopy()
+		cephfsCSIDriver.ObjectMeta.Name = csi.GetCephFSDriverName()
+		if err := csi.CreateCSIDriver(c.ctx, c.Client, cephfsCSIDriver); err != nil {
+			c.log.Error(err, "unable to create cephfs CSIDriver")
+			return ctrl.Result{}, err
+		}
+
+		rbdCSIDriver := templates.RbdCSIDriver.DeepCopy()
+		rbdCSIDriver.ObjectMeta.Name = csi.GetRBDDriverName()
+		if err := csi.CreateCSIDriver(c.ctx, c.Client, rbdCSIDriver); err != nil {
+			c.log.Error(err, "unable to create rbd CSIDriver")
+			return ctrl.Result{}, err
+		}
+
+		prometheusRule := &monitoringv1.PrometheusRule{}
+		if err := k8sYAML.NewYAMLOrJSONDecoder(bytes.NewBufferString(string(pvcPrometheusRules)), 1000).Decode(prometheusRule); err != nil {
+			c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+			return ctrl.Result{}, err
+		}
+
+		prometheusRule.SetNamespace(c.OperatorNamespace)
+
+		err = c.createOrUpdate(prometheusRule, func() error {
+			applyLabels(OperatorConfigMap.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
+			return c.own(prometheusRule, OperatorConfigMap)
+		})
+		if err != nil {
+			c.log.Error(err, "failed to create/update prometheus rules")
+			return ctrl.Result{}, err
+		}
+
+		c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+	}
+	return ctrl.Result{}, nil
+}
+
+func (c *OperatorConfigMapReconciler) deletionPhase() error {
+	if err := c.pvsPresent(); err != nil {
+		c.log.Error(err, "unable to delete ceph resources, PVs consuming client resources are present")
+		return err
+	}
+	cephfsCSIDriverName := csi.GetCephFSDriverName()
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, cephfsCSIDriverName); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete cephfs CSIDriver")
+		return err
+	}
+	rbdCSIDriverName := csi.GetRBDDriverName()
+	if err := csi.DeleteCSIDriver(c.ctx, c.Client, rbdCSIDriverName); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete rbd CSIDriver")
+		return err
+	}
+	if err := c.Client.Delete(c.ctx, c.scc); err != nil && !k8serrors.IsNotFound(err) {
+		c.log.Error(err, "unable to delete SCC")
+		return err
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) pvsPresent() error {
+	// get all the storage class
+	storageClassList := storagev1.StorageClassList{}
+	if err := c.Client.List(c.ctx, &storageClassList); err != nil {
+		return err
+	}
+
+	// create a set of storage class names who are using the ocs-client provisioner
+	// provisioner name prefixed with the namespace name
+	ocsClientStorageClass := make(map[string]bool)
+	for i := range storageClassList.Items {
+		storageClass := &storageClassList.Items[i]
+		if strings.HasPrefix(storageClassList.Items[i].Provisioner, c.OperatorNamespace) {
```

Comment:
the sample code provided in deployer mandates the namespace to be `openshift-storage` and it was correct there but not in here.

remove the prefix func and directly check for these matches
https://github.com/red-hat-storage/ocs-client-operator/blob/18a6a733dff7015d0415dba12061bd467efdff03/pkg/csi/csi.go#L93-L101

these are static funcs, compiler would mostly inline them and no need to worry about storing them again before the loop.